doc/design/katib_on_sqlflow.md

```diff
@@ -0,0 +1,60 @@
+# Katib on SQLFlow
+This design is about optimizing model hyperparameters in SQLFlow by creating Katib step via couler.
+
+## Requirements:
+
+Currently, SQLFlow allows programmers to use SQL queries to enable ML model training, prediction, and explanation. However, hyperparameters optimization is also a necessary function that can help users to tune hyperparamters of their models in an easy way. This function will be supported by Katib in SQLFlow.
+
+## Hyperparameter Optimization in Katib
+
+[Katib](https://github.com/kubeflow/katib) is a Kubernetes Native System for Hyperparameter Tuning and Neural Architecture Search. The system is inspired by Google Vizier and supports multiple ML/DL frameworks (e.g. TensorFlow, Apache MXNet, and PyTorch) and models (e.g., XGBoost).
+
+## Couler and SQLFlow
+
+SQLFlow does not create training jobs on Katib directly. Instead, SQLFlow creates Katib steps in a workflow via couler, and the workflow will be run by Argo on Kubernetes. More details can be found in [couler and SQLFlow](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/couler_sqlflow.md). 
+
+## SQLFlow Syntax
+
+In order to use Katib, programmers need to specify it is a Katib job by `TRAIN katib.{tf, pytorch, xgboost}.{...}` in SQL query. Here we use a simple example to explain the syntax of creating Katib job in SQLFlow.  
+
+``` sql
+SELECT * FROM train_table
+TRAIN katib.xgboost.gbtree
```

Comment:
Please be aware that according to the [model zoo design](https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/model_zoo.md#background), the model names a class in a Docker image and is in the form of `a_data_scientist/regressors:v0.2/MyDNNRegressor`.